src/rotator_library/error_handler.py

```
@@ -619,6 +913,410 @@ def get_retry_after(error: Exception) -> Optional[int]:
+    # Step 3: Determine action based on assessment scope
+    cooldown = max(retry_after or 0, assessment.suggested_cooldown)
+    if cooldown == 0:
+        cooldown = RATE_LIMIT_DEFAULT_COOLDOWN
+
+    if assessment.scope == ThrottleScope.IP:
+        # Multiple credentials throttled - IP-level
+        lib_logger.warning(
+            f"IP-level throttle detected for provider '{provider}' via correlation: "
+            f"{len(assessment.affected_credentials)} credentials affected, "
+            f"confidence={assessment.confidence:.2f}. "
+            f"Blocking provider for {cooldown}s."
```

Ruff (0.15.4)

[warning] Logging statement uses f-string

(G004)